docs/compactor.md

```diff
@@ -0,0 +1,86 @@
+
+But new problems arise with the introduction of compaction - 
+
+Currently full snapshots and the subsequent delta snapshots are grouped under same prefix. When a full snapshot is created, it is placed under a prefix/directory with the name comprising of timestamp. Then subsequent delta snapshots are also pushed into the same directory. Thus each prefix/directory contains a full snapshot and subsequent delta snapshots. So far, it is the job of ETCDBR to start main ETCD process and snapshotter process which takes full snapshot and delta snapshot periodically. But as per our proposal, compaction will be running as parallel process to main ETCD process and snapshotter process. So we can't reliably sync compacted snapshots and subsequent delta snapshots under valid directory/prefix. 
+
+**Current Directory Structure**
+* Backup-192345
+    * Full-Snapshot-192345
+    * Incremental-Snapshot-192355
+    * Incremental-Snapshot-192365
+    * Incremental-Snapshot-192375
+* Backup-192789
+    * Full-Snapshot-192789
+    * Incremental-Snapshot-192799
+    * Incremental-Snapshot-192809
+    * Incremental-Snapshot-192819
+
+To solve the problem, proposal is:
+1. ETCDBR will take the first full snapshot after it starts main ETCD Process and snapshotter process. But after taking the first full snapshot, snapshotter will only continue taking delta snapshots but no further full snapshots.
+2. Flatten the directory structure of backup folder. Save all the full snapshots, delta snapshots and compacted snapshots under same directory/prefix. Also, name snapshots after revision number instead of timestamps. Restorer will restore from compacted snapshots that have highest revision number in the name and delta snapshots that have higher revision numbers in name than the comapcted full snapshot.
+
+**Proposed Directory Structure**
+* Backup
+    * Full-Snapshot-revision-1
+    * Incremental-Snapshot-revision-2
+    * Incremental-Snapshot-revision-7
+    * Incremental-Snapshot-revision-13
+    * Incremental-Snapshot-revision-15 (It containes command upto revision 18)
+    * Full-Snapshot-revision-18 (Compacted)
+    * Incremental-Snapshot-revision-19
+    * Incremental-Snapshot-revision-23
+    * Incremental-Snapshot-revision-28
+    * Incremental-Snapshot-revision-33 (It containes command upto revision 38)
+    * Full-Snapshot-revision-38 (Compacted)
+    * Incremental-Snapshot-revision-39
+    * Incremental-Snapshot-revision-42
+    * Incremental-Snapshot-revision-43
+    * Incremental-Snapshot-revision-49
+
+3. A meta-data file might need to be added in the Backup directory to mark which compacted snapshot is ready to be used for restoration. Restorer would study the meta-data file and pickup the ready full snapshot from there and apply all the delta snapshots having revision number greater than that. So, with the backup directory as above, if the content of metadata file is like following:
+
+* Full-Snapshot-revision-1
+* Full-Snapshot-revision-18
```

Comment:
Should we go for a json or yaml structure here?

Reply:
Used yaml